### ISSUE TITLE
ghost busters

### ISSUE THREAD

**Issue by ctm** [ctm/mb2-doc#415]
Fix the code that computes which pots a player is all-in on, so that busts and bounties work.

Yesterday we played two bounty tournaments and in more cases than not the bounty wasn't awarded because my code does not recognize certain classes of all-ins as being all-in for the pot in question.  I more or less know what the issue is and how to fix it.

**Comment by ctm** [comment]
Looks like the culprit was some code I put in for the sevens rule.

Specifically when we're creating the pots we run through all the actions and see if any contain forfeited actions (which is really pretty rare since they can only occur in CA Lowball and even then people rarely check a seven, at least on this site ...). When the action and forfeited action  are the same as the current pot, we add the amount of the action into the current pot. If they different we split off a new pot. The code that was adding the amount was throwing away whatever all-in information we had already computed.

So, the action associated with all-ins was being handled fine, but mb2 was forgetting who was all-in.. Later after all the candidates' hands have been evaluated, we use that all-in info from the various pots to figure out who bust whom, but ... some of that info was thrown away, so... we punted.

The fix was simple. I'm going to leave a bunch of debug logging in place for this evening's tournament and tonight's tournament, just in case more things go wrong. I'll leave this issue open until--at minimum--the finish of this evening's tournament.

**Comment by ctm** [comment]
My fix wasn't complete and may have even made things worse. Upon inspection I see there are basically two places where I needed to do a particular thing and I was only doing them in one. To rub salt in the wounds, the place where I was doing it wasn't even the "correct" place to do it, so I've removed it from there and put it in the other two places.

I spent a little time trying to figure. way I could unify things and only do it in one place, but I couldn't find a way to do it that didn't make the rest of the code even more convoluted.

So, I've smoke-tested my fix and am deploying now, so it can be tested in the night game.
